src/popup/router/pages/Send.vue

```diff
@@ -197,27 +198,46 @@ export default {
     },
     async fetchFee() {
       await this.$watchUntilTruly(() => this.sdk);
-      this.fee = await calculateFee(
-        this.token === 0 ? TX_TYPES.txSign : TX_TYPES.contractCall,
-        this.sdk.Ae.defaults,
-      );
+      this.fee = calculateFee(!this.selectedToken ? TX_TYPES.txSign : TX_TYPES.contractCall, {
+        ...this.sdk.Ae.defaults,
+        ...(this.selectedToken && {
+          callerId: this.account.publicKey,
+          contractId: this.selectedToken.contract,
+        }),
+      });
     },
     setTxDetails(tx) {
+      if (tx.tx.type === 'ContractCallTx') {
+        this.successTx.amount = convertToken(
+          tx.amount,
+          -this.availableTokens[tx.contractId].decimals,
+        );
+        this.successTx.token = tx.contractId;
+        this.successTx.to = tx.recipientId;
+        this.successTx.from = tx.callerId;
+        this.successTx.hash = tx.hash;
+        return;
+      }
       this.successTx.amount = parseFloat(tx.tx.amount / 10 ** 18).toFixed(3);
       this.successTx.to = tx.tx.recipientId;
       this.successTx.from = tx.tx.senderId;
       this.successTx.hash = tx.hash;
     },
     async send() {
-      const amount = aeToAettos(this.form.amount);
+      const amount = !this.selectedToken
+        ? aeToAettos(this.form.amount)
+        : convertToken(this.form.amount, this.selectedToken.decimals);
       const receiver = this.form.address;
       const calculatedMaxValue = this.balance > this.fee ? this.balance - this.fee : 0;
       let errorModalType = '';
       if (receiver === '' || (!checkAddress(receiver) && !chekAensName(receiver))) {
         errorModalType = 'incorrect-address';
       }
       if (this.form.amount <= 0) errorModalType = 'incorrect-amount';
-      if (calculatedMaxValue - this.form.amount <= 0 && this.token === 0) {
+      if (
+        (calculatedMaxValue - this.form.amount <= 0 && !this.selectedToken) ||
+        (this.selectedToken && this.selectedToken.convertedBalance < this.form.amount)
```

Comment:
I'm trying to send tokens with 0 ae balance, but I don't see the `insufficient-balance` modal. I should see it because I don't have enough ae to do that.